speech_to_text/speech_to_text.livemd

```diff
@@ -0,0 +1,217 @@
+# Speech to text
+
+```elixir
+Logger.configure(level: :info)
+
+Mix.install(
+  [
+    :bumblebee,
+    :exla,
+    :membrane_core,
+    :membrane_portaudio_plugin
+  ],
+  config: [
+    nx: [default_backend: EXLA.Backend]
+  ]
+)
+```
+
+## Prerequisites
+
+On Linux, you need to install PortAudio, for example with `sudo apt install portaudio19-dev`.
+
+## Introduction
+
+This livebook example shows how to perform a real-time speech-to-text conversion with the use of the [Membrane Framework](https://github.com/membraneframework) and the [Bumblebee](https://github.com/elixir-nx/bumblebee).
+
+You will see how to fetch the audio from your microphone, perform preprocessing and create your own Membrane element that runs an AI speech-to-text conversion model.
+
+## Element performing a speech to text
+
+We need to write a custom sink that will perform speech-to-text and display the resulting transcription. Let's call this element `SpeechToText`.
+
+To perform the transcription the element will use the Open AI's [Whisper](https://openai.com/research/whisper) model. 
+It can be easily loaded and used with a "little" help from [Bumblebee](https://github.com/elixir-nx/bumblebee).
+
+The Whipser model requires the input audio samples to be in `f32le` format, which means, that they are represented as floating numbers written on 32 bits, with little endian bytes order. The required input sample rate is 16 kHz, as well as only single-channel samples are allowed.
+
+### Initialization
+
+In the initialization process, we load the `Whisper` model, along with the featurizer, tokenizer, and generation configuration. Then we create a speech-to-text serving and indicate that we want to use `EXLA` backend for the tensor operations.
+
+### Buffers handling
+
+Once a buffer arrives, we check if it contains enough amount of audio and accumulate it in the state otherwise. Then we perform basic VAD (Voice Activity Detection) to remove silent chunks, what improves the model behaviour. Then we convert the audio to an Nx tensor, feed the model and parse output, which is then sent to via the output pad.
```

Comment:
```suggestion
Once a buffer arrives, we check if it contains enough amount of audio and accumulate it in the state otherwise. Then we perform basic VAD (Voice Activity Detection) to remove silent chunks, which improves the model behaviour. Then we convert the audio to an Nx tensor, feed the model, and parse the output, which is then sent via the output pad.
```